share/eds/eds.go

```diff
@@ -0,0 +1,221 @@
+		Version: 1,
+	}, w.w)
+}
+
+// writeQuadrants reorders the shares to quadrant order and writes them to the CARv1 file.
+func (w *writingSession) writeQuadrants() error {
+	shares := quadrantOrder(w.eds)
+	for _, share := range shares {
+		cid, err := ipld.CidFromNamespacedSha256(nmt.Sha256Namespace8FlaggedLeaf(share))
+		if err != nil {
+			return fmt.Errorf("failure to get cid from share: %w", err)
+		}
+		err = util.LdWrite(w.w, cid.Bytes(), share)
+		if err != nil {
+			return fmt.Errorf("failure to write share: %w", err)
+		}
+	}
+	return nil
+}
+
+// writeProofs iterates over the in-memory blockstore's keys and writes all inner nodes to the CARv1 file.
+func (w *writingSession) writeProofs(ctx context.Context) error {
+	// we only stored proofs to the store, so we can just iterate over them here without getting any leaves
+	proofs, err := w.store.AllKeysChan(ctx)
+	if err != nil {
+		return fmt.Errorf("failure to get all keys from the blockstore: %w", err)
+	}
+	for proofCid := range proofs {
+		node, err := w.store.Get(ctx, proofCid)
+		if err != nil {
+			return fmt.Errorf("failure to get proof from the blockstore: %w", err)
+		}
+		cid, err := ipld.CidFromNamespacedSha256(nmt.Sha256Namespace8FlaggedInner(node.RawData()))
+		if err != nil {
+			return fmt.Errorf("failure to get cid: %w", err)
+		}
+		err = util.LdWrite(w.w, cid.Bytes(), node.RawData())
+		if err != nil {
+			return fmt.Errorf("failure to write proof to the car: %w", err)
+		}
+	}
+	return nil
+}
+
+// quadrantOrder reorders the shares in the EDS to quadrant row-by-row order, prepending the respective namespace
+// to the shares.
+// e.g. [ Q1 R1 | Q1 R2 | Q1 R3 | Q1 R4 | Q2 R1 | Q2 R2 .... ]
+func quadrantOrder(eds *rsmt2d.ExtendedDataSquare) [][]byte {
+	size := eds.Width() * eds.Width()
+	shares := make([][]byte, size)
+
+	quadrantWidth := int(eds.Width() / 2)
+	quadrantSize := quadrantWidth * quadrantWidth
+	for i := 0; i < quadrantWidth; i++ {
+		for j := 0; j < quadrantWidth; j++ {
+			cells := getQuadrantCells(eds, uint(i), uint(j))
+			innerOffset := i*quadrantWidth + j
+			for quadrant := 0; quadrant < 4; quadrant++ {
+				shares[(quadrant*quadrantSize)+innerOffset] = prependNamespace(quadrant, cells[quadrant])
+			}
+		}
+	}
+	return shares
+}
+
+// getQuadrantCells returns the cell of each EDS quadrant with the passed inner-quadrant coordinates
+func getQuadrantCells(eds *rsmt2d.ExtendedDataSquare, i, j uint) [][]byte {
+	cells := make([][]byte, 4)
+	quadrantWidth := eds.Width() / 2
+	cells[0] = eds.GetCell(i, j)
+	cells[1] = eds.GetCell(i, j+quadrantWidth)
+	cells[2] = eds.GetCell(i+quadrantWidth, j)
+	cells[3] = eds.GetCell(i+quadrantWidth, j+quadrantWidth)
+	return cells
+}
+
+// prependNamespace adds the namespace to the passed share if in the first quadrant,
+// otherwise it adds the ParitySharesNamespace to the beginning.
+func prependNamespace(quadrant int, share []byte) []byte {
+	switch quadrant {
+	case 0:
+		return append(share[:8], share...)
```

Comment:
Lets use const instead of number here.
```suggestion
		return append(share[:appconsts.NamespaceSize], share...)
```

Reply:
its already done in ReadEDS so im going to resolve here